src/dsp_tools/models/value.py

```diff
@@ -23,14 +23,21 @@ def __init__(self, xmlstr: str) -> None:
     def __str__(self) -> str:
         return self.__xmlstr
 
+    def replace(self, fromStr: str, toStr: str) -> None:
+        self.__xmlstr = self.__xmlstr.replace(fromStr, toStr)
+
     def get_all_iris(self) -> Optional[list[str]]:
         return self.__iriregexp.findall(self.__xmlstr)
 
-    def replace(self, fromStr: str, toStr: str) -> None:
-        self.__xmlstr = self.__xmlstr.replace(fromStr, toStr)
+    def find_all_iri_in_xmlstr(self) -> set[str]:
+        return set(regex.findall(pattern='href="IRI:(.*?):IRI"', string=self.__xmlstr))
 
-    def regex_replace(self, pattern: str, repl: str) -> None:
-        self.__xmlstr = regex.sub(pattern=repr(pattern)[1:-1], repl=repl, string=self.__xmlstr)
+    def replace_one_internal_id_with_iri_in_string(self, internal_id: str, iri: str) -> None:
```

Comment:
The method name doesn't mention that the replacement happens only in salsah-links. I think `replace_id_with_iri_in_salsah_links()` would be clearer.